### ISSUE TITLE
VCF - Mismatch AS/SP in headers doesn't fall back

### ISSUE THREAD

**Issue by keiranmraine** [cancerit/cgpPindel#62]
If the BAM headers are mismatched for AS/SP a warning should be emitted and fall back to command line option accepted as a resolution.

```
ERROR: Trying to merge contigs with conflicting data:
$VAR1 = bless( {
                 '_checksum' => undef,
                 '_length' => '249250621',
                 '_species' => 'Human',
                 '_assembly' => 'NCBI37',
                 '_name' => '1'
               }, 'Sanger::CGP::Vcf::Contig' );
 
$VAR1 = bless( {
                 '_checksum' => undef,
                 '_length' => 249250621,
                 '_species' => 'human',
                 '_assembly' => 'GRCh37d5',
                 '_name' => '1'
               }, 'Sanger::CGP::Vcf::Contig' );
at /opt/wtsi-cgp/bin/pindel_2_combined_vcf.pl line 70.
```

**Comment by keiranmraine** [comment]
Looks like the `$contig->compare()` method will have to be dropped in favour of a manual check of `-name` and `-length`.

The alternative is to alter cgpVcf, passing options to ignore species/build mismatches (behaviour should only change when they are defined).

https://github.com/cancerit/cgpVcf/blob/bd2ba581c6392cce01a79f0df8eff43d13fedba7/lib/Sanger/CGP/Vcf/Contig.pm#L84-L107

**But** if checksum is absent in one-input _that should pass_, if it's present in both it should fail if a mismatch.

@sb43 what are your thoughts?  Caveman doesn't use the function in cgpVcf

**Comment by sb43** [comment]
@keiranmraine I think modifying cgpVCF  is ideal option, I will look into it.

**Comment by sb43** [comment]
As discussed with @keiranmraine  adding separate functions to compare length & name and assembly & species will be added to pindel code base.

**Comment by sb43** [comment]
Fixed in v3.0.6
